akka-actor-typed-tests/src/test/scala/akka/actor/typed/WidenSpec.scala

```diff
@@ -123,14 +123,58 @@ class WidenSpec extends ScalaTestWithActorTestKit("""
         }
 
       EventFilter[ActorInitializationException](occurrences = 1).intercept {
-        val ref = spawn(widen(widen(Behaviors.receiveMessage[String] { message =>
+        val ref = spawn(widen(widen(Behaviors.receiveMessage[String] { _ =>
           Behaviors.same
         })))
 
         probe.expectTerminated(ref, 3.seconds)
       }
 
     }
+
+    "be possible to combine with inner timers" in {
+      val probe = TestProbe[String]()
+      val behv = Behaviors
+        .withTimers[String] { timers =>
+          timers.startSingleTimer("timer", "a", 10.millis)
+          Behaviors.receiveMessage { msg =>
+            probe.ref ! msg
+            Behaviors.same
+          }
+        }
+        .widen[String] {
+          case msg => msg.toUpperCase()
+        }
+
+      val ref = spawn(behv)
+
+      probe.expectMessage("A")
+
+      ref ! "b"
+      probe.expectMessage("B")
+    }
+
+    "be possible to combine with outer timers" in {
+      val probe = TestProbe[String]()
+      val behv = Behaviors.withTimers[String] { timers =>
+        timers.startSingleTimer("timer", "a", 10.millis)
+        Behaviors
+          .receiveMessage[String] { msg =>
+            probe.ref ! msg
+            Behaviors.same
+          }
+          .widen[String] {
+            case msg => msg.toUpperCase()
+          }
+      }
```

Comment:
Unrelated: as an API this would be nice to simplify if we can at some point
```scala
Behaviors.withA[T] { a => 
  a.doSomething() 
  Behaviors.withB[T] { b =>
    /* do something */
    Behaviors.same
  }
}
```

Reply:
What do you mean? It's a good way to add needed capabilities in scope.

Reply:
True, scoped makes sense. It's the nesting redundant declarations of `Behaviors.*` that struck me.

Reply:
Behaviors. is the goto factory for all things. Will quickly become familiar, and if one is looking for reducing characters import Behaviors._ is an option.